crowbar_framework/app/models/service_object.rb

```diff
@@ -719,7 +719,7 @@ def validate_one_for_role(proposal, role)
   def validate_at_least_n_for_role(proposal, role, n)
     elements = proposal["deployment"][@bc_name]["elements"]
 
-    if not elements.has_key?(role) or elements[role].length < n
+    if not elements.key?(role) or elements[role].length < n
```

Comment:
Style/AndOr: Use `||` instead of `or`. (https://github.com/bbatsov/ruby-style-guide#no-and-or-or)<br>Style/Not: Use `!` instead of `not`. (https://github.com/bbatsov/ruby-style-guide#bang-not-not)